docs/guide/migration.md

```diff
@@ -81,6 +81,8 @@ If you prefer to stick with `style.css` like in Vite 5, you can set `build.lib.c
 
 There are other breaking changes which only affect few users.
 
+- [[#18240] refactor: use `originalFileNames/names`](https://github.com/vitejs/vite/pull/18240):
+  - any plugin that emits assets with type `asset` must include the `names` array in the asset to avoid breaking the build when the manifest is enabled. Older versions of Vite use the `name` field to check whether the asset should be included in the manifest.
```

Comment:
It's not just older versions. Vite 6 does too

https://github.com/vitejs/vite/blob/b39e69638b3e2e658ff6712be83b549b28103c3d/packages/vite/src/node/plugins/asset.ts#L400

Reply:
maybe only when using `generateBundle` hook adding the asset to the bundle?